Design/reflectometry-idf.md

```diff
@@ -0,0 +1,62 @@
+
+## Solution
+
+This task should be thought of in terms of four main sub-tasks:
+
+  ### 1. POLREF IDF support for relative pixel map
+  
+  Andrew currently has a scan of the POLREF detector which he uses in a script to move each detector pixel into place every time 
+  they run a reduction. This appears to be something which can be supported in an IDF by obtaining the relative pixel positions 
+  from the scan and hard-coding them. The actual specular pixel position could then be obtained from the logs and used to do this 
+  rotation into place from within the IDF.
+  
+  ### 2. Support multiple L-distances through a complex beam-path
+  
+  Current workarounds include the following:
+    - Separate 'neutronic' and 'physical instrument' IDFs
+	- Dynamic modification of positions via `MoveInstrumentComponent`
+	- Storage of the L1 & L2 distances in a separate table, which `ConvertUnits` can be made to 'understand'
+	
+  The ideal solution to this task would be to create a new `beam-path` tag for description within the IDF. This can then can be used 
+  with Owen’s [Instrument 2.0 prototype](https://github.com/DMSC-Instrument-Data/instrument-prototype) which specifically deals with 
+  the in-memory side of this task. He has mentioned that ray tracing will not be possible, and even with the prototype discussed above 
+  only one flight path per component has been considered, so additional thought will need to be given to multiple flight paths as this 
+  is required by the reflectometers in some configurations.
+  
+  >That is to say, that if you have a mirror, the prototype discussed does not treat different areas of the surface as yielding a 
+  >different flight path (even though that is reality). There may also be wavelength dependence on the flight path to consider. 
+  >The other thing to consider will be what precision of flight path is needed.
+  
+  ### 3. Describe multiple modes of configuration
+  
+  The instrument scientists would like to describe multiple modes of configuration (NR, PNR liquids, etc.) within a single IDF. 
+  This will require a sensible way to describe multiple degrees of freedom and, if done accurately, will help in solving an issue they 
+  are having with Q error bars and some other possible source of systematic error. It will also allow them to update their workflow in 
+  real time as the report often having to change configurations with some users which ultimately takes a lot of time.
+  
+  A possible solution to this would be to have a generic IDF which can obtain the specified configuration through the log files delivered 
+  by SECI, as is currently done in some IDFs. The IDF would then be able to update accordingly through a switch-statement of sorts, 
+  moving the relevant components into place. No such log currently exists and so if this solution is decided upon it will require
+  collaboration with the IBEX team.
```

Comment:
Other solutions based on delivering the geometry with the data:

1. The IDF could potentially be written into the raw nexus file. I believe since LoadInstrument is used, it would automatically process that. This would be a change on the ECP side (SECI) but I prefer it to adding logs as that's just one additional step that could break along the way.
2. Move to use the NexusGeometry Format in the nexus files. This would be a larger change on the SECI /IBEX side but would automatically solve these problems. This should be discussed in conjunction with the NexusConstructor with @matthew-d-jones